dashboard/lib/api/v1/pd/workshop_score_summarizer.rb

```diff
@@ -59,150 +59,199 @@ module Api::V1::Pd::WorkshopScoreSummarizer
     :how_clearly_presented_s,
     :how_interesting_s,
     :how_often_given_feedback_s,
-    :help_quality_s,
     :how_comfortable_asking_questions_s,
     :how_often_taught_new_things_s,
     :things_facilitator_did_well_s,
     :things_facilitator_could_improve_s
   ].freeze
 
-  def get_score_for_workshops(workshops, facilitator_breakdown: false, include_free_responses: false)
-    report_rows = Hash.new(0).merge({number_teachers: 0, response_count: 0})
+  FACILITATOR_SPECIFIC_MULTIPLE_CHOICE_QUESTIONS = FACILITATOR_SPECIFIC_QUESTIONS - FREE_RESPONSE_QUESTIONS
+  FACILITATOR_SPECIFIC_FREE_RESPONSE_QUESTIONS = FACILITATOR_SPECIFIC_QUESTIONS & FREE_RESPONSE_QUESTIONS
 
-    response_count = 0
+  def generate_summary_report(workshop: nil, workshops:, course:, facilitator_name: nil, facilitator_breakdown: false)
+    survey_report = Hash.new
 
-    facilitator_scores = nil
-    free_responses = Hash.new
+    if workshop
+      survey_report[:this_workshop] = get_score_for_workshops(
+        workshops: [workshop],
+        include_free_responses: true,
+        facilitator_name_filter: facilitator_name
+      )
+    end
+
+    survey_report[:all_my_workshops_for_course] = get_score_for_workshops(
+      workshops: workshops,
+      include_free_responses: false,
+      facilitator_name_filter: facilitator_name
+    )
+
+    aggregate_for_all_workshops = JSON.parse(AWS::S3.download_from_bucket('pd-workshop-surveys', "aggregate-workshop-scores-#{CDO.rack_env}"))
+    survey_report[:all_workshops_for_course] = aggregate_for_all_workshops[course].try(&:symbolize_keys) || {}
 
     if facilitator_breakdown
-      facilitator_scores = Hash.new
+      facilitators = workshops.flat_map(&:facilitators).sort.uniq
 
-      workshops.flat_map(&:facilitators).uniq.each do |facilitator|
-        facilitator_scores[facilitator.name] = Hash.new(0)
+      facilitators.each do |facilitator|
+        survey_report[facilitator.name] = get_score_for_workshops(
+          workshops: workshops.facilitated_by(facilitator),
+          include_free_responses: false,
+          facilitator_name_filter: facilitator.name
+        )
       end
     end
 
-    if include_free_responses
-      FREE_RESPONSE_QUESTIONS.each do |question|
-        free_responses[question] = []
-      end
-    end
+    survey_report
+  end
 
-    workshops.each do |workshop|
-      enrollment_ids = workshop.enrollments.pluck(:id)
+  def get_score_for_workshops(workshops:, include_free_responses:, facilitator_name_filter:)
+    facilitators = workshops.flat_map(&:facilitators).map(&:name)
```

Comment:
does this need a `.uniq`? What if we have multiple workshops for a given facilitator?

Reply:
Will add